test/lint/git-subtree-check.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2017-2022 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# Checks that a certain prefix is pure subtree, and optionally whether the
+# referenced commit is present in any fetched remote.
+
+import argparse
+import sys
+import subprocess
+
+def parse_args():
+    """Parse command line arguments"""
+    parser = argparse.ArgumentParser (
+        description="""
+            Checks that a certain prefix is pure subtree, and optionally whether the
+            referenced commit is present in any fetched remote.
+        """
+    )
+
+    parser.add_argument('dir', help="The prefix within the repository to check.")
+    parser.add_argument('commit', nargs='?', default='HEAD', help="The commit to check, if it is not provided, HEAD will be used.")
+    parser.add_argument("--repository", "-r", action='store_true', required=False, help="Check that subtree commit is present in repository. To do this check, fetch the subtreed remote first. (e.g. git fetch https://github.com/bitcoin-core/secp256k1.git\ntest/lint/git-subtree-check.sh -r src/secp256k1)")
+
+    return parser.parse_args()
+
+def find_latest_squash(dir, commit):
+    sq = main = sub = None
+    stdout = subprocess.check_output(["git", "log", "--grep=^git-subtree-dir: " + dir + "/*$", "--pretty=format: START %H%n%s%n%n%b%nEND%n", commit], universal_newlines=True, encoding="utf8")
+    for line in stdout.splitlines():
+        if "START" in line:
+            sq = line.lstrip().split(' ')[1]
+        if line.startswith("git-subtree-mainline:"):
+            main = line.split(':', 2)[1].split(' ')
+        if line.startswith("git-subtree-split:"):
+            sub = line.split(':', 2)[1].split(' ')
+        if "END" in line:
+            if sub:
+                if main:
+                    sq = sub
+                return(f"{sq} {sub}")
+            sq = main = sub = None
+
+def main():
+    args = parse_args()
+
+    # find the latest subtree update
+    latest_squash = find_latest_squash(args.dir.rstrip('/'), args.commit)
+    if not latest_squash:
+        print(f"Error: {args.dir} is not a subtree")
+        sys.exit(2)
+    else:
+        old = latest_squash.split(' ')[0]
+        rev = latest_squash.split(' ')[1]
```

Comment:
tiny nit: you can avoid putting `else` if you're exiting in the `if` branch:
```
    if not latest_squash:
        print(f"Error: {args.dir} is not a subtree")
        sys.exit(2)
    old = latest_squash.split(' ')[0]
    rev = latest_squash.split(' ')[1]
```
This reduces indentation, but here it's basically irrelevant :)

Reply:
Very good point. I thought it helped with readability a bit but can easily go through and remove the else statements after exiting

Reply:
As you wish, if you prefer to leave it like that it's ok for me :)